test/extended/federation.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+#
+# Runs the federation tests against a standard started server.
+source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
+source "${OS_ROOT}/test/extended/setup.sh"
+
+os::test::extended::setup
+os::util::ensure::built_binary_exists kubefed
+os::test::extended::focus "$@"
+
+export OS_FEDERATION_NAME="${OS_FEDERATION_NAME:-origin-federation}"
+export OS_FEDERATION_NAMESPACE="${OS_FEDERATION_NAMESPACE:-federation-system}"
+export OS_FEDERATION_SERVICE_ACCOUNT="system:serviceaccount:${OS_FEDERATION_NAMESPACE}:default"
+
+os::test::junit::declare_suite_start "extended/federation"
+
+os::log::info "Deploying federation control plane"
+os::cmd::expect_success "kubefed init '${OS_FEDERATION_NAME}' --dns-provider=google-clouddns \
+        --federation-system-namespace='${OS_FEDERATION_NAMESPACE}' \
+        --etcd-persistent-storage=false \
+        --image='openshift/origin-federation:v1.5.0-0.alpha.3.1554.a0c834a'"  # TODO Remove
+
+# Ensure the controller manager will be able to access cluster configuration stored as
+# secrets in the federation namespace.
+os::log::debug "Granting the admin role to the service account of the federation namespace"
```

Comment:
it would be nice if we can create a namespace before calling kubefed so we can pre-grant this.

https://github.com/kubernetes/kubernetes/issues/47159